api/routes/comments.js

```diff
@@ -0,0 +1,112 @@
+const express = require('express');
+const User = require('../models/User');
+const Post = require('../models/Post');
+const Comment = require('../models/Comment');
+const authenticated = require('../middleware/withAuth');
+
+const app = express();
+
+//Create a new comment on a post 
+app.post("/post/comment", authenticated, (req, res) => {
+	const commentContent = req.body.comment;
+
+	if (!commentContent) {
+		res.status(400).send({
+		  message: "You should write a comment!",
+		});
+		return;
+	  }
+	//The postId and the userId will be already available on the front-end in the form for before a user wants to create a comment
+	//The need to be send in the body, but we are not going to show them on the page.
+	const postId = req.body.postId;
+	const userId = req.body.userId;
+	
+	//Comment object
+    const comment = {
+        comment_content: commentContent,
+		post_id: postId,
+		user_id: userId
+    };
+	//Creating the comment and saving to the database
+    Comment.create(comment)
+		.then((data) => {
+			console.log(data);
+			res.status(201).send(data);
+		})
+		.catch((err) => {
+			res.status(500).send({
+				message: err.message || 'Some error occurred while creating the Comment.',
+			});
+		});
+
+});
+
+//Update a comment
+app.put("/post/comment/:id", authenticated, (req, res) => {
+	let commentId = req.params.id;
+	//console.log(commentId);
+	const userId = req.body.userId
+	if(!commentId) {
+	 res.status(400).send({
+		 message: 'No comment id provided in the params'
+	 });
+	 return;
+	}
+	//The idea here is to show update/delete small buttons only on the owner's comment/comments.
+	//There won't be an option for a user to click update/delete on the comments from other users.
+	
+	const newComment = req.body.comment;
+	const where = { where: {comment_id: Number(commentId), user_id: userId} };
+	   //console.log(newComment);
+	   //console.log(where);
+
+	//Updating the comment in the database
+	Comment.update(
+		{comment_content: newComment},
+		 where
+	)
+		.then((data) => {
+			res.json(data).status(200);
```

Comment:
can you add this here in case some errors

>  if (Boolean(data[0]) === false) {
> 				res.status(401).send({
> 					status: 401,
> 					message: 'You cant change this comment',
> 				});
> 				return;
> 			}